### PR TITLE
feat: system tray custom menu & podman modal

### PR DESCRIPTION
**Pull request by corn-potage**
- added new podman modal to handle various podman status cases from sidebar and 
- added custom menu to handle node statuses
- clicking on node opens the correct node in the app
- quit menu opens dialog
- fixed various existing podman related bugs
- added warning on requirements step for users who currently use Mac OS lower than 13.0.0, as virtualization doesn't work for lower versions

**will need to be tested on Windows and Linux**

Notes:
- currently not a react component but javascript creating html nodes in the custom window; had difficulty and will need more time to figure that out
- couldn't get colored notification icon to work
- need more time to figure out a better way to handle granular node statuses such as "Initializing", "Syncing", etc so it can be used the same way in Tray, NodePackageScreen, etc.

#526

### CODE REVIEW

**Comment by jgresham** [comment]
Looking great! and themed! 
Screenshot 2024-06-03 at 2 53 46 PM

File: src/main/main.ts [hunk 1]
```diff
@@ -99,6 +99,19 @@ const getAssetPath = (...paths: string[]): string => {
   return path.join(RESOURCES_PATH, ...paths);
 };
 
+const getTrayPath = (...paths: string[]): string => {
+  return path.join(
+    __dirname,
+    '..',
+    '..',
+    'src',
```
Comment (jgresham) [inline src/main/main.ts:107]: I think this path will not work in packaged builds

**Comment by corn-potage** [comment]
@jgresham Ahh I added podman modal to Add Node in the beginning for debug purposes and forgot to revert back. Will fix 👍

I have the following for the tray:
```
if (process.env.NODE_ENV === 'development') {
    trayWindow?.webContents.openDevTools();
  }
```
should I remove this entirely?

Do you happen to have Windows? Would you be able to see if the custom menu shows up fine in the right position? I'll need to check on Linux too..

**Comment by jgresham** [comment]
Nice set of features and fixes in here!

**Comment by jgresham** [comment]
todo for me is to stop all nodes and podman machine before the "full quit" from the tray icon. right now this isn't true
Screenshot 2024-06-05 at 10 14 07 AM

**Pull request merged**